.github/workflows/xyz-build.yml

```diff
@@ -0,0 +1,260 @@
+name: XYZ Build
+
+env:
+  DEFAULT_NOTES: ""
+
+on:
+  push:
+    branches:
+      - main
+      - xyz-env
+    paths:
+      - .github/workflows/xyz-build.yml
+      - cloud-function/**
```

Comment:
Not sure how I feel about this - if we use xyz as a more permanent staging for feature branches, it could be a little confusing to have them overwritten by a merged PR modifying a file in one of these paths. That said - it's not catastrophic, one can always re-deploy from the feature branch.